ecosystem/typescript/sdk_v2/src/bcs/move-types/structs.ts

```diff
@@ -0,0 +1,398 @@
+import { Serializable, Serializer } from "../serializer";
+import { Deserializable, Deserializer } from "../deserializer";
+import { Bool, U128, U16, U256, U32, U64, U8 } from "./primitives";
+import { AnyNumber, HexInput } from "../../types";
+import { AccountAddress } from "../../core";
+
+/**
+ * This class is the Aptos Typescript SDK representation of a Move `vector<T>`,
+ * where `T` represents either a primitive type (`bool`, `u8`, `u64`, ...)
+ * or a BCS-serializable struct itself.
+ *
+ * It is a BCS-serializable, array-like type that contains an array of values of type `T`,
+ * where `T` is a class that implements `Serializable`.
+ *
+ * The purpose of this class is to facilitate easy construction of BCS-serializable
+ * Move `vector<T>` types.
+ *
+ * @example
+ * // in Move: `vector<u8> [1, 2, 3, 4];`
+ * const vecOfU8s = new Vector<U8>([new U8(1), new U8(2), new U8(3), new U8(4)]);
```

Comment:
so I tried to do this initially, but realized ultimately it's impossible to infer the type and thus serialize it correctly, because you can't use generic type info at runtime.

So I tried both `new Vector([1, 2, 3, 4], U8)` and a few other things, but ultimately realized that this is perfect for a factory method, which is in there:

`Vector.U8([1, 2, 3, 4])`

Reply:
I played around with enforcing homogenous types when instantiating a vector so you could get strongly typed vectors and not `Vector<Serializable>` but it resulted in other issues (you have to create a `NoUnion<T>` type and figured I'd think about it later